Documentation/ApiOverview/Database/ExpressionBuilder/_RepositoryLeftPad.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Domain\Repository;
+
+use TYPO3\CMS\Core\Database\ConnectionPool;
+
+final class MyTableRepository
+{
+    private const TABLE_NAME = 'tt_content';
+    public function __construct(private readonly ConnectionPool $connectionPool) {}
+
+    public function demonstrateLeftPad(): void
+    {
+        $queryBuilder = $this->connectionPool
+            ->getQueryBuilderForTable(self::TABLE_NAME);
+        $expression1 = $queryBuilder->expr()->leftPad(
+            $queryBuilder->quote('123'),
+            10,
+            '0',
+        );
+
+        $expression2 = $queryBuilder->expr()->leftPad(
+            $queryBuilder->expr()->castVarchar($queryBuilder->quoteIdentifier('uid')),
+            10,
+            '0',
+        );
+
+        $expression3 = $queryBuilder->expr()->leftPad(
+            $queryBuilder->expr()->concat(
+                $queryBuilder->quote('1'),
+                $queryBuilder->quote('2'),
+                $queryBuilder->quote('3'),
+            ),
+            10,
+            '0',
+        );
+
+        $expression4 = $queryBuilder->expr()->leftPad(
```

Comment:
What does this do? From my understanding it would return

"00( 1123 )"

? Or is "( 1123 )" somehow just cast to "1123" and then results in "0000001123"? Would need a small code comment to indicate what this achieves.

Reply:
I just copied it over from the changelog

Reply:
This is actually a duplicate casting, resulting in this SQL:

```
LPAD((CAST((CAST(( 1123 ) AS CHAR(255))) AS CHAR(255))), (CAST(10 AS SIGNED INTEGER)), '0') 1
```

It seems to want to showcase a custom sub-expression, so I just leave it at that.